images/conduit/install.sh

```diff
@@ -0,0 +1,31 @@
+#!/usr/bin/env bash
+
+# Build conduit and put binary in /tmp.
+#
+# Configured with environment variables:
+#   URL    - Git repository URL.
+#   BRANCH - Git branch to clone.
+#   SHA    - (optional) Specific commit hash to checkout.
+set -e
+
+START=$(date "+%s")
+THIS=$(basename "$0")
+
+echo "$THIS: starting conduit image install"
+
+# Sometimes conduit is disabled, detect the missing build config.
+if [ -z "${BRANCH}" ] || [ -z "${URL}" ]; then
+  echo "Missing BRANCH or URL environment variable. Skipping install."
+  # returning with 0 because it will run the disabled server.
+  exit 0
+fi
+
+git clone --single-branch --branch "${BRANCH}" "${URL}" /opt/conduit
```

Comment:
How does that work when checking out the commit hash below? I might have tried this in the past and `--single-branch` was the best I could find.

Reply:
I guess the answer is simple... clone at `depth 1` if there is no `SHA`, otherwise clone the whole thing. I'm not sure it's worth the extra code complexity.

Reply:
Yeah, I forgot about that.